payment/src/main/java/org/killbill/billing/payment/core/sm/control/OperationControlCallback.java

```diff
@@ -99,16 +100,14 @@ public OperationResult doOperationCallback() throws OperationException {
                                                                                                      paymentStateControlContext.isApiPayment(),
                                                                                                      paymentStateContext.getCallContext());
 
-                final PriorPaymentControlResult pluginResult;
                 try {
-                    pluginResult = executePluginPriorCalls(paymentStateControlContext.getPaymentControlPluginNames(), paymentControlContext);
-                    if (pluginResult != null && pluginResult.isAborted()) {
-                        // Transition to ABORTED
-                        return PluginDispatcher.createPluginDispatcherReturnType(OperationResult.EXCEPTION);
-                    }
+                    executePluginPriorCalls(paymentStateControlContext.getPaymentControlPluginNames(), paymentControlContext);
                 } catch (final PaymentControlApiException e) {
                     // Transition to ABORTED and throw PaymentControlApiException to caller.
                     throw new OperationException(e, OperationResult.EXCEPTION);
+                } catch (final PaymentApiException paymentAbortedException) {
+                    // Transition to ABORTED
+                    throw new OperationException(paymentAbortedException, OperationResult.EXCEPTION);
```

Comment:
One option here is to throw create a specialized `PaymentControlApiException` for the abort case.
The code here would change to:
```java
try {
    executePluginPriorCalls(paymentStateControlContext.getPaymentControlPluginNames(), paymentControlContext);
} catch (final PaymentControlApiAbortException e) {
    // Transition to ABORTED
    final PaymentApiException paymentAbortedException = new PaymentApiException(ErrorCode. PAYMENT_PLUGIN_API_ABORTED, e.getPluginName());
    throw new OperationException(paymentAbortedException, OperationResult.EXCEPTION);
} catch (final PaymentControlApiException e) {
    // Transition to ABORTED and throw PaymentControlApiException to caller.
    throw new OperationException(e, OperationResult.EXCEPTION);
}